src/business_logic/transaction/fee.rs

```diff
@@ -0,0 +1,124 @@
+use super::error::TransactionError;
+use crate::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            objects::{CallInfo, TransactionExecutionContext},
+        },
+        fact_state::{
+            in_memory_state_reader::InMemoryStateReader, state::ExecutionResourcesManager,
+        },
+        state::cached_state::CachedState,
+    },
+    definitions::general_config::StarknetGeneralConfig,
+    services::api::contract_class::EntryPointType,
+};
+use felt::{felt_str, Felt};
+use num_traits::ToPrimitive;
+use std::collections::HashMap;
+
+// ----------------------------------------------------------------------------
+/// Transfers the amount actual_fee from the caller account to the sequencer.
+/// Returns the resulting CallInfo of the transfer call.
+
+pub(crate) fn execute_fee_transfer(
+    state: &mut CachedState<InMemoryStateReader>,
+    general_config: StarknetGeneralConfig,
+    tx_context: TransactionExecutionContext,
+    actual_fee: u64,
+) -> Result<CallInfo, TransactionError> {
+    if actual_fee as u64 > tx_context.max_fee {
+        return Err(TransactionError::FeeError(
+            "Actual fee exceeded max fee.".to_string(),
+        ));
+    }
+
+    let fee_token_address = general_config.starknet_os_config.fee_token_address.clone();
+
+    let calldata = [
+        general_config.block_info.sequencer_address.0.clone(),
+        Felt::from(actual_fee),
+        0.into(),
+    ]
+    .to_vec();
+
+    // Value generated from `get_selector_from_name('TRANSFER_ENTRY_POINT_SELECTOR')`.
+    let entry_point_selector =
+        felt_str!("232670485425082704932579856502088130646006032362877466777181098476241604910");
+
+    let fee_transfer_call = ExecutionEntryPoint::new(
```

Comment:
I dont quite understand what contract you are executing here @mmsc2

Reply:
I understand that it should call the `transactions.cairo` program in order to execute validations.

Reply:
There's a comment of mine earlier suggesting placing a comment with the source of the selector, which in that case is `transfer`.

Reply:
I like @azteca1998´s idea 
@mmsc2 please add that :)

Reply:
there is a comment above the function indicating the source, i will change the selector name to 'transfer' for simplicity.